### PR TITLE
add follow ups

### PR DESCRIPTION
**Pull request by sarahmarciano**

### CODE REVIEW

File: js/components/TicketDetails/FollowUps.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    iconStyle: {
+        paddingRight: 10
+    },
+    dateStyle: {
+        fontWeight: 'bold',
+        paddingBottom: 4
+    }
+});
+export default FollowUps;
```
Comment (raphaelboukara) [inline js/components/TicketDetails/FollowUps.js]: I think you can simplify all this stuff:

1- Create a `FollowUp` component in `FollowUp.js` for example:
```js
function FollowUp({ creation, content }) {
    const { followupItem, followupTitle, iconStyle, dateStyle } = styles;

    return (
        <View style={followupItem}>
            <View style={followupTitle}>
                <Icon name='chat-bubble' color={primaryColor} iconStyle={iconStyle} size={20} />
                <Text style={dateStyle}>{moment(creation).format('D/M/YYYY')}</Text>
            </View>
            <Text>{content}</Text>
        </View>
    );
}

FollowUp.propTypes = {
    creation: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
};

const mapStateToProps = ({ followUps }, { id }) => {
    const { creation, content } = followUps[id];

    return { creation, content };
};

//if followUp is in fact an usual comment
const mapStateToProps = ({ comments }, { id }) => {
    const { creation, content } = comments[id];

    return { creation, content };
};

export default ReactRedux.connect(mapStateToProps)(FollowUp);
```

2- Create a `FollowUps` component in `FollowUps.js` for example:
```js
function FollowUps({ displayFollowUps, ids, ticketId }) {
    const { title, viewStyle } = styles;

    return displayFollowUps && (
        <View style={viewStyle}>
            <Text style={title}>{I18n.t('followUps')}</Text>
            {ids.map((id) => <FollowUp key={id} id={id}/>)}
            <AddFollowUp ticketId={ticketId} />
        </View>
    );
};

FollowUps.propTypes = {
    ids: PropTypes.array.isRequired,
    displayFollowUps: PropTypes.bool.isRequired,
    ticketId: PropTypes.number.isRequired
};

const mapStateToProps = (state, { ticketId }) => ({
    ids: state.tickets[ticketId].followUps, //I don't know the format of the state
    displayFollowUps: state.user.canAddFollowUp, //Something like this
    ticketId
});

export default ReactRedux.connect(mapStateToProps)(FollowUps);
```

Comment (raphaelboukara) [reply js/components/TicketDetails/FollowUps.js]: You need to adapt this code to your state but I think you understand the idea. One component for the followUps, one for the followUp. 
- Connected component followUps receive a ticket id, `mapStateToProps` via the ticketId will build the data.
- Connect component followUp receive an id, `mapStateToProps` via the id will return the followUp data.

Comment (raphaelboukara) [reply js/components/TicketDetails/FollowUps.js]: think simple :)

File: js/redux/reducers/TicketsDetailsReducer.js [hunk 2]
```diff
@@ -16,6 +18,23 @@ export default (state = INITIAL_STATE, action) => {
...
+            return { ...state, [meta.id]: { ...state[meta.id], followUpLoading: true } };
+        case `${ADD_FOLLOW_UP}_FULFILLED`:
+            console.log(payload);
+            return {
+                ...state,
+                [meta.id]: {
+                    ...state[meta.id],
+                    followUpLoading: false,
+                    canAddFollowup: false,
+                    followups: [{ content: payload.data.Content, creation: payload.data.Creaion }, ...state[meta.id].followups]
```
Comment (raphaelboukara) [inline js/redux/reducers/TicketsDetailsReducer.js]: warning: typo

File: js/components/TicketDetails/AddFollowUp.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    render() {
+        const { onAddFollowUp, ticketId, followUpLoading } = this.props;
+        const { followup, followUpStyle } = styles;
+        const { comment } = this.state;
+        const isDisabled = comment === '' || followUpLoading;
+        return (
+            <View>
+                <View style={followUpStyle}>
+                    <Item floatingLabel>
```
[the remark this answers is not in the document]
Comment (sarahmarciano) [reply js/components/TicketDetails/AddFollowUp.js]: @raphaelboukara yes, this is a native base component which can have his label floating up when the field is focused
see http://docs.nativebase.io/Components.html#floating-label-headref

File: js/components/TicketDetails/AddFollowUp.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                        <Label>{I18n.t('leaveComment')}</Label>
+                        <Input
+                            value={comment}
+                            onChangeText={(text) => this.setState({ comment: text })}
+                        />
+                    </Item>
+                    <ButtonLoading
```
[the remark this answers is not in the document]
Comment (sarahmarciano) [reply js/components/TicketDetails/AddFollowUp.js]: @raphaelboukara This is a custom button that we've made that show a spinner when he has his loading property set to true we use it when doing an asynchronous request

File: js/components/TicketDetails/AddFollowUp.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        disabled={isDisabled}
+                        loading={followUpLoading}
+                        onPress={() => onAddFollowUp(ticketId, comment)}
```
[the remark this answers is not in the document]
Comment (sarahmarciano) [reply js/components/TicketDetails/AddFollowUp.js]: @raphaelboukara yes indeed... it's better

**Pull request merged**